superset/reports/commands/execute.py

```diff
@@ -0,0 +1,256 @@
+        image_url = self._get_url(report_schedule, user_friendly=True)
+        user = security_manager.find_user(app.config["THUMBNAIL_SELENIUM_USER"])
+        image_data = screenshot.compute_and_cache(
+            user=user, cache=thumbnail_cache, force=True,
+        )
+        if not image_data:
+            raise ReportScheduleScreenshotFailedError()
+        return ScreenshotData(url=image_url, image=image_data)
+
+    def _get_notification_content(
+        self, report_schedule: ReportSchedule
+    ) -> NotificationContent:
+        """
+        Gets a notification content, this is composed by a title and a screenshot
+        :raises: ReportScheduleScreenshotFailedError
+        """
+        screenshot_data = self._get_screenshot(report_schedule)
+        if report_schedule.chart:
+            name = report_schedule.chart.slice_name
+        else:
+            name = report_schedule.dashboard.dashboard_title
+        return NotificationContent(name=name, screenshot=screenshot_data)
+
+    def _send(self, report_schedule: ReportSchedule) -> None:
+        """
+        Creates the notification content and sends them to all recipients
+
+        :raises: ReportScheduleNotificationError
+        """
+        notification_errors = []
+        notification_content = self._get_notification_content(report_schedule)
+        for recipient in report_schedule.recipients:
+            notification = create_notification(recipient, notification_content)
+            try:
+                notification.send()
+            except NotificationError as ex:
+                # collect notification errors but keep processing them
+                notification_errors.append(str(ex))
+        if notification_errors:
+            raise ReportScheduleNotificationError(";".join(notification_errors))
+
+    def run(self) -> None:
+        with session_scope(nullpool=True) as session:
+            try:
+                start_dttm = datetime.utcnow()
+                self.validate(session=session)
+                if not self._model:
+                    raise ReportScheduleExecuteUnexpectedError()
+                self.set_state_and_log(session, start_dttm, ReportLogState.WORKING)
+                # If it's an alert check if the alert is triggered
+                if self._model.type == ReportScheduleType.ALERT:
+                    if not AlertCommand(self._model).run():
+                        self.set_state_and_log(session, start_dttm, ReportLogState.NOOP)
+                        return
+
+                self._send(self._model)
+
+                # Log, state and TS
+                self.set_state_and_log(session, start_dttm, ReportLogState.SUCCESS)
+            except ReportScheduleAlertGracePeriodError as ex:
+                self.set_state_and_log(
+                    session, start_dttm, ReportLogState.NOOP, error_message=str(ex)
+                )
+            except ReportSchedulePreviousWorkingError as ex:
```

Comment:
should this one be retried ?

Reply:
I think not, since I'm proposing a 1min beat, the beat cycle itself is the retry, yet the cron for the alert can be set to 1h or 10min etc. Undecided..., I propose not to throw too much logic here, unless we find it's something we really should add

Reply:
sounds good